src/libclang/preprocessor.cpp

```diff
@@ -1184,9 +1185,16 @@ detail::preprocessor_output detail::preprocess(const libclang_compile_config& co
             {
                 if (first_line.try_reset() && lm.value().file == path && lm.value().line == 1u)
                 {
+                    std::string xpath;
```

Comment:
Can you move this computation to the top of the function, so it only happens once?

Reply:
moved to the top